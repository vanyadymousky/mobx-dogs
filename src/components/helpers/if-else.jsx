import { Children } from 'react';
import PropTypes from 'prop-types';
import { ifElse, always, head, last, take } from 'ramda';

/**
 * Based on condition property returns either first element or second
 * @param children
 * @param condition
 * @returns {*}
 */
export const IfElse = ({ children, condition }) => ifElse(
  always(condition),
  head,
  last,
)(take(2, Children.toArray(children)));

IfElse.propTypes = {
  children: PropTypes.arrayOf(PropTypes.node).isRequired,
  condition: PropTypes.bool.isRequired,
};
