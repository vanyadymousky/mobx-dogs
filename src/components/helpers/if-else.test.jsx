import { IfElse } from './if-else';


describe('<IfElse />', () => {
  let wrapper;

  describe('when condition is true', () => {
    const condition = true;

    test('should render first element', () => {
      wrapper = shallow(<IfElse condition={condition}><span>first</span><span>second</span></IfElse>);

      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('when condition is false', () => {
    const condition = false;

    test('should render second element', () => {
      wrapper = shallow(<IfElse condition={condition}><span>first</span><span>second</span></IfElse>);

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when children length is more than 2', () => {
    const condition = false;

    test('should ignore 3,4,5... elements', () => {
      wrapper = shallow(<IfElse condition={condition}><span>first</span><span>second</span><span>third</span></IfElse>);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
