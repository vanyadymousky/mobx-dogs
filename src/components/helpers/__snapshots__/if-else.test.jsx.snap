// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IfElse /> when children length is more than 2 should ignore 3,4,5... elements 1`] = `
<span
  key=".1"
>
  second
</span>
`;

exports[`<IfElse /> when condition is false should render second element 1`] = `
<span
  key=".1"
>
  second
</span>
`;

exports[`<IfElse /> when condition is true should render first element 1`] = `
<span
  key=".0"
>
  first
</span>
`;
